libraries/plugins/account_history/account_history_plugin.cpp

```diff
@@ -212,9 +215,25 @@ void account_history_plugin_impl::add_account_history( const account_id_type acc
        obj.most_recent_op = ath.id;
        obj.total_ops = ath.sequence;
    });
-   // remove the earliest account history entry if too many
-   // _max_ops_per_account is guaranteed to be non-zero outside
-   if( stats_obj.total_ops - stats_obj.removed_ops > _max_ops_per_account )
+   // Amount of history to keep depends on if account is in the "extended history" list
+   bool extended_hist = false;
+   for ( auto eh_account_id : _extended_history_accounts ) {
+      extended_hist |= (account_id == eh_account_id);
+   }
+   if ( _extended_history_registrars.size() > 0 ) {
+      const account_id_type registrar_id = account_id(db).registrar;
```

Comment:
Querying DB once on every operation is a bit inefficient. Perhaps better create a new index in the plugin to save the tracked accounts. Why a new index? Because the list is dynamic and grows over time.

Reply:
Although the `account_index` is big, the complexity of querying it by ID is linear, so the performance hit is probably not significant. Best if we can have some data to show the differences E.G. time spent for replaying with the old code v.s. the new code.